{"expand":"renderedFields,names,schema,transitions,operations,editmeta,changelog","id":"12377237","self":"https://issues.apache.org/jira/rest/api/latest/issue/12377237","key":"DERBY-3035","fields":{"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false},"timespent":null,"project":{"self":"https://issues.apache.org/jira/rest/api/2/project/10594","id":"10594","key":"DERBY","name":"Derby","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/projectavatar?pid=10594&avatarId=10122","24x24":"https://issues.apache.org/jira/secure/projectavatar?size=small&pid=10594&avatarId=10122","16x16":"https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=10594&avatarId=10122","32x32":"https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=10594&avatarId=10122"},"projectCategory":{"self":"https://issues.apache.org/jira/rest/api/2/projectCategory/10090","id":"10090","description":"DB related projects","name":"DB"}},"fixVersions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313143","id":"12313143","description":"Head of 10.3 branch","name":"10.3.3.1","archived":false,"released":false},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313401","id":"12313401","description":"Head of 10.4 branch","name":"10.4.2.1","archived":false,"released":false},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313771","id":"12313771","description":"build: 764942, 14/Apr/09. Voted release: 28/Apr/09. Announced: 1/May/09","name":"10.5.1.1","archived":false,"released":true,"releaseDate":"2009-04-28"}],"aggregatetimespent":null,"resolution":{"self":"https://issues.apache.org/jira/rest/api/2/resolution/1","id":"1","description":"A fix for this issue is checked into the tree and tested.","name":"Fixed"},"customfield_12310220":"2007-11-20 10:03:19.28","customfield_12312322":null,"customfield_12312323":null,"customfield_12310420":"23388","customfield_12310222":"1_*:*_1_*:*_38121039144_*|*_5_*:*_1_*:*_0","customfield_12312320":null,"customfield_12312321":null,"customfield_12312120":null,"customfield_12312121":null,"resolutiondate":"2008-11-14T18:28:10.197+0000","workratio":-1,"customfield_12312328":null,"customfield_12312329":null,"customfield_12312326":null,"customfield_12310300":null,"customfield_12312327":null,"customfield_12312324":null,"customfield_12312325":null,"lastViewed":null,"watches":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-3035/watchers","watchCount":1,"isWatching":false},"created":"2007-08-31T13:17:31.053+0000","priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.png","name":"Major","id":"3"},"labels":[],"customfield_12312333":null,"customfield_12312334":null,"customfield_12310310":"2.0","customfield_12312331":null,"customfield_12312332":null,"timeestimate":null,"aggregatetimeoriginalestimate":null,"versions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12312590","id":"12312590","description":"","name":"10.3.1.4","archived":false,"released":true,"releaseDate":"2007-08-10"}],"customfield_12311120":null,"customfield_12312330":null,"issuelinks":[],"customfield_12312339":null,"assignee":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"customfield_12312337":null,"customfield_12312338":null,"updated":"2011-01-21T17:50:48.398+0000","customfield_12312335":null,"customfield_12311720":null,"customfield_12312336":null,"status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"components":[{"self":"https://issues.apache.org/jira/rest/api/2/component/11412","id":"11412","name":"Store"}],"timeoriginalestimate":null,"description":"I tried to restore a database backup from a backup directory by using the following connection url\n\n    jdbc:derby:mydb;restoreFrom=mybackup\n\nThe follwoing error occured\n\njava.sql.SQLException: Failed to start database 'mydb', see the next exception for details.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(Unknown Source)\n\tat org.apache.derby.impl.jdbc.Util.newEmbedSQLException(Unknown Source)\n\tat org.apache.derby.impl.jdbc.Util.seeNextException(Unknown Source)\n\tat org.apache.derby.impl.jdbc.EmbedConnection.bootDatabase(Unknown Source)\n\tat org.apache.derby.impl.jdbc.EmbedConnection.<init>(Unknown Source)\n\tat org.apache.derby.impl.jdbc.EmbedConnection30.<init>(Unknown Source)\n\tat org.apache.derby.impl.jdbc.EmbedConnection40.<init>(Unknown Source)\n\tat org.apache.derby.jdbc.Driver40.getNewEmbedConnection(Unknown Source)\n\tat org.apache.derby.jdbc.InternalDriver.connect(Unknown Source)\n\tat org.apache.derby.jdbc.EmbeddedDriver.connect(Unknown Source)\n\tat net.sourceforge.squirrel_sql.fw.sql.SQLDriverManager.getConnection(SQLDriverManager.java:133)\n\tat net.sourceforge.squirrel_sql.client.mainframe.action.OpenConnectionCommand.execute(OpenConnectionCommand.java:97)\n\tat net.sourceforge.squirrel_sql.client.mainframe.action.ConnectToAliasCommand$SheetHandler.run(ConnectToAliasCommand.java:283)\n\tat net.sourceforge.squirrel_sql.fw.util.TaskExecuter.run(TaskExecuter.java:82)\n\tat java.lang.Thread.run(Unknown Source)\nCaused by: java.sql.SQLException: Failed to start database 'mydb', see the next exception for details.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(Unknown Source)\n\t... 15 more\nCaused by: java.sql.SQLException: Error copying file (during backup) from mybackup\\BACKUP.HISTORY to mydb\\BACKUP.HISTORY.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(Unknown Source)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(Unknown Source)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(Unknown Source)\n\tat org.apache.derby.impl.jdbc.Util.generateCsSQLException(Unknown Source)\n\t... 12 more\nCaused by: ERROR XSRS5: Error copying file (during backup) from mybackup\\BACKUP.HISTORY to mydb\\BACKUP.HISTORY.\n\tat org.apache.derby.iapi.error.StandardException.newException(Unknown Source)\n\tat org.apache.derby.impl.store.raw.RawStore.restoreRemainingFromBackup(Unknown Source)\n\tat org.apache.derby.impl.store.raw.RawStore.boot(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.BaseMonitor.boot(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.TopService.bootModule(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.BaseMonitor.startModule(Unknown Source)\n\tat org.apache.derby.iapi.services.monitor.Monitor.bootServiceModule(Unknown Source)\n\tat org.apache.derby.impl.store.access.RAMAccessManager.boot(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.BaseMonitor.boot(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.TopService.bootModule(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.BaseMonitor.startModule(Unknown Source)\n\tat org.apache.derby.iapi.services.monitor.Monitor.bootServiceModule(Unknown Source)\n\tat org.apache.derby.impl.db.BasicDatabase.bootStore(Unknown Source)\n\tat org.apache.derby.impl.db.BasicDatabase.boot(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.BaseMonitor.boot(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.TopService.bootModule(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.BaseMonitor.bootService(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.BaseMonitor.startProviderService(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.BaseMonitor.findProviderAndStartService(Unknown Source)\n\tat org.apache.derby.impl.services.monitor.BaseMonitor.startPersistentService(Unknown Source)\n\tat org.apache.derby.iapi.services.monitor.Monitor.startPersistentService(Unknown Source)\n\t... 12 more\n\n\nI wonder why the file BACKUP.HISTORY is so importent that it blows the whole import process if it is not there. \nI do not have this file because i retrieved the backup by performing an offline backup. In this case there is no BACKUP.HISTORY file.\n\nIf i create a empty file with the name BACKUP.HISTORY in the backup folder, everything works fine.\n","customfield_10010":null,"timetracking":{},"customfield_12312026":null,"customfield_12310200":null,"customfield_12312023":null,"customfield_12312024":null,"customfield_12312340":null,"aggregatetimeestimate":null,"customfield_12312022":null,"customfield_12312341":null,"customfield_12312220":null,"customfield_12310921":null,"customfield_12310920":"37395","summary":"Cannot restore backups without BACKUP.HISTORY file","creator":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=s_huber","name":"s_huber","emailAddress":"huste at vr-web dot de","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Stefan Huber","active":true},"subtasks":[],"customfield_12310090":[{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10102","value":"Patch Available","id":"10102"}],"reporter":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=s_huber","name":"s_huber","emailAddress":"huste at vr-web dot de","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Stefan Huber","active":true},"customfield_12310291":null,"customfield_12310290":null,"aggregateprogress":{"progress":0,"total":0},"environment":"Windows XP","customfield_12311020":null,"customfield_12310050":{"self":"https://issues.apache.org/jira/rest/api/2/customFieldOption/10052","value":"Normal","id":"10052"},"duedate":null,"customfield_12310250":null,"progress":{"progress":0,"total":0},"comment":{"startAt":0,"maxResults":7,"total":7,"comments":[{"self":"https://issues.apache.org/jira/rest/api/2/issue/12377237/comment/12543834","id":"12543834","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kirkwylie","name":"kirkwylie","emailAddress":"kirk at kirkwylie dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kirk Wylie","active":true},"body":"This also affects you if you use createFrom=true semantics (to bootstrap a database from another database).\r\n\r\nInterestingly, if I create a brand new database, this file is never present.\r\n\r\nThis is a massive regression in functionality for our environment. Thanks for the empty file workaround, but this is pretty serious.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kirkwylie","name":"kirkwylie","emailAddress":"kirk at kirkwylie dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kirk Wylie","active":true},"created":"2007-11-20T10:03:19.280+0000","updated":"2007-11-20T10:03:19.280+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12377237/comment/12644180","id":"12644180","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"body":"Here is the debug stack trace.\r\n\r\nij> connect 'jdbc:derby:wombat;create=true';\r\nij> connect 'jdbc:derby:wombat2;createFrom=wombat';\r\nERROR XJ040: Failed to start database 'wombat2', see the next exception for details.\r\njava.sql.SQLException: Failed to start database 'wombat2', see the next exception for details.\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:95)\r\n        at org.apache.derby.impl.jdbc.Util.newEmbedSQLException(Util.java:87)\r\n        at org.apache.derby.impl.jdbc.Util.seeNextException(Util.java:223)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection.bootDatabase(EmbedConnection.java:2620)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection.<init>(EmbedConnection.java:374)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection30.<init>(EmbedConnection30.java:73)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection40.<init>(EmbedConnection40.java:54)\r\n        at org.apache.derby.jdbc.Driver40.getNewEmbedConnection(Driver40.java:68)\r\n        at org.apache.derby.jdbc.InternalDriver.connect(InternalDriver.java:238)\r\n        at org.apache.derby.jdbc.AutoloadedDriver.connect(AutoloadedDriver.java:119)\r\n        at java.sql.DriverManager.getConnection(DriverManager.java:582)\r\n        at java.sql.DriverManager.getConnection(DriverManager.java:154)\r\n        at org.apache.derby.impl.tools.ij.ij.dynamicConnection(ij.java:1325)\r\n        at org.apache.derby.impl.tools.ij.ij.ConnectStatement(ij.java:1175)\r\n        at org.apache.derby.impl.tools.ij.ij.ijStatement(ij.java:1003)\r\n        at org.apache.derby.impl.tools.ij.utilMain.runScriptGuts(utilMain.java:328)\r\n        at org.apache.derby.impl.tools.ij.utilMain.go(utilMain.java:248)\r\n        at org.apache.derby.impl.tools.ij.Main.go(Main.java:215)\r\n        at org.apache.derby.impl.tools.ij.Main.mainCore(Main.java:181)\r\n        at org.apache.derby.impl.tools.ij.Main.main(Main.java:73)\r\n        at org.apache.derby.tools.ij.main(ij.java:59)\r\nCaused by: java.sql.SQLException: Failed to start database 'wombat2', see the next exception for details.\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java\r\n9)\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)\r\n        ... 20 more\r\nCaused by: java.sql.SQLException: Error copying file (during backup) from wombat\\BACKUP.HISTORY to C:\\kmarsden\\repro\\\r\nby-3035\\wombat2\\BACKUP.HISTORY.\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java\r\n9)\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)\r\n        at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Util.java:201)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection.bootDatabase(EmbedConnection.java:2614)\r\n        ... 17 more\r\nCaused by: ERROR XSRS5: Error copying file (during backup) from wombat\\BACKUP.HISTORY to C:\\kmarsden\\repro\\derby-3035\r\nmbat2\\BACKUP.HISTORY.\r\n        at org.apache.derby.iapi.error.StandardException.newException(StandardException.java:303)\r\n        at org.apache.derby.impl.store.raw.RawStore.restoreRemainingFromBackup(RawStore.java:1145)\r\n        at org.apache.derby.impl.store.raw.RawStore.boot(RawStore.java:238)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.boot(BaseMonitor.java:2019)\r\n        at org.apache.derby.impl.services.monitor.TopService.bootModule(TopService.java:291)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.startModule(BaseMonitor.java:573)\r\n        at org.apache.derby.iapi.services.monitor.Monitor.bootServiceModule(Monitor.java:427)\r\n        at org.apache.derby.impl.store.access.RAMAccessManager.boot(RAMAccessManager.java:1019)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.boot(BaseMonitor.java:2019)\r\n        at org.apache.derby.impl.services.monitor.TopService.bootModule(TopService.java:291)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.startModule(BaseMonitor.java:573)\r\n        at org.apache.derby.iapi.services.monitor.Monitor.bootServiceModule(Monitor.java:427)\r\n        at org.apache.derby.impl.db.BasicDatabase.bootStore(BasicDatabase.java:780)\r\n        at org.apache.derby.impl.db.BasicDatabase.boot(BasicDatabase.java:196)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.boot(BaseMonitor.java:2019)\r\n        at org.apache.derby.impl.services.monitor.TopService.bootModule(TopService.java:291)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.bootService(BaseMonitor.java:1856)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.startProviderService(BaseMonitor.java:1722)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.findProviderAndStartService(BaseMonitor.java:1602)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.startPersistentService(BaseMonitor.java:1021)\r\n        at org.apache.derby.iapi.services.monitor.Monitor.startPersistentService(Monitor.java:550)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection.bootDatabase(EmbedConnection.java:2581)\r\n        ... 17 more\r\nERROR XSRS5: Error copying file (during backup) from wombat\\BACKUP.HISTORY to C:\\kmarsden\\repro\\derby-3035\\wombat2\\BA\r\nP.HISTORY.\r\njava.sql.SQLException: Error copying file (during backup) from wombat\\BACKUP.HISTORY to C:\\kmarsden\\repro\\derby-3035\\\r\nbat2\\BACKUP.HISTORY.\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:95)\r\n        at org.apache.derby.impl.jdbc.Util.generateCsSQLException(Util.java:201)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection.bootDatabase(EmbedConnection.java:2614)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection.<init>(EmbedConnection.java:374)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection30.<init>(EmbedConnection30.java:73)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection40.<init>(EmbedConnection40.java:54)\r\n        at org.apache.derby.jdbc.Driver40.getNewEmbedConnection(Driver40.java:68)\r\n        at org.apache.derby.jdbc.InternalDriver.connect(InternalDriver.java:238)\r\n        at org.apache.derby.jdbc.AutoloadedDriver.connect(AutoloadedDriver.java:119)\r\n        at java.sql.DriverManager.getConnection(DriverManager.java:582)\r\n        at java.sql.DriverManager.getConnection(DriverManager.java:154)\r\n        at org.apache.derby.impl.tools.ij.ij.dynamicConnection(ij.java:1325)\r\n        at org.apache.derby.impl.tools.ij.ij.ConnectStatement(ij.java:1175)\r\n        at org.apache.derby.impl.tools.ij.ij.ijStatement(ij.java:1003)\r\n        at org.apache.derby.impl.tools.ij.utilMain.runScriptGuts(utilMain.java:328)\r\n        at org.apache.derby.impl.tools.ij.utilMain.go(utilMain.java:248)\r\n        at org.apache.derby.impl.tools.ij.Main.go(Main.java:215)\r\n        at org.apache.derby.impl.tools.ij.Main.mainCore(Main.java:181)\r\n        at org.apache.derby.impl.tools.ij.Main.main(Main.java:73)\r\n        at org.apache.derby.tools.ij.main(ij.java:59)\r\nCaused by: java.sql.SQLException: Error copying file (during backup) from wombat\\BACKUP.HISTORY to C:\\kmarsden\\repro\\\r\nby-3035\\wombat2\\BACKUP.HISTORY.\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java\r\n9)\r\n        at org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)\r\n        ... 19 more\r\nCaused by: ERROR XSRS5: Error copying file (during backup) from wombat\\BACKUP.HISTORY to C:\\kmarsden\\repro\\derby-3035\r\nmbat2\\BACKUP.HISTORY.\r\n        at org.apache.derby.iapi.error.StandardException.newException(StandardException.java:303)\r\n        at org.apache.derby.impl.store.raw.RawStore.restoreRemainingFromBackup(RawStore.java:1145)\r\n        at org.apache.derby.impl.store.raw.RawStore.boot(RawStore.java:238)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.boot(BaseMonitor.java:2019)\r\n        at org.apache.derby.impl.services.monitor.TopService.bootModule(TopService.java:291)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.startModule(BaseMonitor.java:573)\r\n        at org.apache.derby.iapi.services.monitor.Monitor.bootServiceModule(Monitor.java:427)\r\n        at org.apache.derby.impl.store.access.RAMAccessManager.boot(RAMAccessManager.java:1019)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.boot(BaseMonitor.java:2019)\r\n        at org.apache.derby.impl.services.monitor.TopService.bootModule(TopService.java:291)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.startModule(BaseMonitor.java:573)\r\n        at org.apache.derby.iapi.services.monitor.Monitor.bootServiceModule(Monitor.java:427)\r\n        at org.apache.derby.impl.db.BasicDatabase.bootStore(BasicDatabase.java:780)\r\n        at org.apache.derby.impl.db.BasicDatabase.boot(BasicDatabase.java:196)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.boot(BaseMonitor.java:2019)\r\n        at org.apache.derby.impl.services.monitor.TopService.bootModule(TopService.java:291)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.bootService(BaseMonitor.java:1856)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.startProviderService(BaseMonitor.java:1722)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.findProviderAndStartService(BaseMonitor.java:1602)\r\n        at org.apache.derby.impl.services.monitor.BaseMonitor.startPersistentService(BaseMonitor.java:1021)\r\n        at org.apache.derby.iapi.services.monitor.Monitor.startPersistentService(Monitor.java:550)\r\n        at org.apache.derby.impl.jdbc.EmbedConnection.bootDatabase(EmbedConnection.java:2581)\r\n        ... 17 more\r\nij>","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"created":"2008-10-31T00:34:04.570+0000","updated":"2008-10-31T00:34:04.570+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12377237/comment/12644383","id":"12644383","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"body":"The fix seems as simple as not trying to copy the history file if it does not exist.  Attached is a patch that does that.  I still need to add tests and run tests, but this is the approach I plan to take.  Let me know if that might cause any problems.\r\n\r\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"created":"2008-10-31T18:13:06.631+0000","updated":"2008-10-31T18:13:06.631+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12377237/comment/12644390","id":"12644390","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bryanpendleton","name":"bryanpendleton","emailAddress":"bpendleton dot derby at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Bryan Pendleton","active":true},"body":"If the backup is missing the history file, should the results of the restore be to:\r\n - leave the restored database with no history file?\r\n - leave the restored database with a brand-new empty history file?\r\n - leave the restored database with whatever history file was present\r\n   prior to doing the restore?\r\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bryanpendleton","name":"bryanpendleton","emailAddress":"bpendleton dot derby at gmail dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Bryan Pendleton","active":true},"created":"2008-10-31T18:21:36.323+0000","updated":"2008-10-31T18:21:36.323+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12377237/comment/12644392","id":"12644392","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"body":"The patch I propose will leave whatever history file was present prior to doing the restore or none if there was none.\r\nI think this is right, because the current comments say:\r\n\r\n\r\n\t\t// if this is a roll-forward recovery, backup history file \r\n\t\t// will already there in the database and will be the latest \r\n\t\t// copy; if it exists, do not copy from backup\r\n\r\nKathey\r\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"created":"2008-10-31T18:29:45.621+0000","updated":"2008-10-31T18:29:45.621+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12377237/comment/12645291","id":"12645291","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"body":"Attached is a patch for this issue, which will check for the existence of the backup history file before it tries to copy it.  Added a test OfflineBackupTest. Please review.\r\n\r\n\r\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"created":"2008-11-05T18:54:03.950+0000","updated":"2008-11-05T18:54:03.950+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12377237/comment/12645801","id":"12645801","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"body":"If there are no comments. I'll check this in Monday  (derby-3035_diff.txt)\r\n\r\nThanks\r\n\r\nKathey\r\n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kmarsden","name":"kmarsden","emailAddress":"kmarsdenderby at sbcglobal dot net","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Kathey Marsden","active":true},"created":"2008-11-07T15:59:05.479+0000","updated":"2008-11-07T15:59:05.479+0000"}]},"votes":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-3035/votes","votes":1,"hasVoted":false},"worklog":{"startAt":0,"maxResults":20,"total":0,"worklogs":[]},"customfield_12311820":"0|i06rjb:"}}